cliconfig/config.go

```diff
@@ -29,20 +31,15 @@ var (
 	configDir = os.Getenv("DOCKER_CONFIG")
 )
 
-func getDefaultConfigDir(confFile string) string {
-	confDir := filepath.Join(homedir.Get(), confFile)
-	// if the directory doesn't exist, maybe we called docker with sudo
-	if _, err := os.Stat(configDir); err != nil {
-		if os.IsNotExist(err) {
-			return filepath.Join(homedir.GetWithSudoUser(), confFile)
-		}
-	}
-	return confDir
-}
-
 func init() {
 	if configDir == "" {
-		configDir = getDefaultConfigDir(".docker")
+		configDir = filepath.Join(homedir.Get(), defaultConfigDir)
+		if _, err := os.Stat(configDir); err != nil {
+			// if the directory doesn't exist, maybe we called docker with sudo
```

Comment:
alright, I see.

> Also, back to my other question... do other tools act this way?

Honestly I don't know - but the use is being able to resolve to the *wanted* user directory when running sudo.
I guess a solution w/o even touching docker is to run:
```
sudo -u runcom docker pull registryauth.com:5000/busybox
```
which correctly resolves to my home dir.